### ISSUE TITLE
qt-mod build fails on darwin

### ISSUE THREAD

**Issue by davidak** [NixOS/nixpkgs#37777]
## Issue description

```
building path(s) ‘/nix/store/h12xgvwy1h5pbdds463kbv69vjrksqmd-qt-mod-4.8.7-dev’, ‘/nix/store/m5fj1pd1jw7l35ambd4kcld8ci870rii-qt-mod-4.8.7’
unpacking sources
unpacking source archive /nix/store/rplan8ayri93d638pm9cavd7kbw71myr-source
source root is source
patching sources
applying patch /nix/store/9lvggxlccy24g59khxzas2yy16sdlhy4-glib-2.32.patch
patching file src/3rdparty/javascriptcore/JavaScriptCore/wtf/Threading.h
Hunk #1 succeeded at 83 (offset -6 lines).
applying patch /nix/store/yba8b3rcs8h9zw3w1a6ldlm9xp3bz619-libressl.patch
patching file src/network/ssl/qsslsocket_openssl_symbols.cpp
patching file src/network/ssl/qsslsocket_openssl.cpp
applying patch /nix/store/nq7zhnwb06m7qam53z3695p3xw0dgxdv-dlopen-absolute-paths.diff
patching file src/corelib/tools/qlocale_icu.cpp
Hunk #1 succeeded at 85 (offset 4 lines).
Hunk #2 succeeded at 115 (offset 4 lines).
patching file src/gui/painting/qcups.cpp
patching file src/gui/painting/qprinterinfo_unix.cpp
patching file src/network/kernel/qhostinfo_unix.cpp
patching file src/plugins/platforms/xcb/qglxintegration.cpp
patching file src/plugins/platforms/xlib/qglxintegration.cpp
patching file src/plugins/platforms/xlib/qxlibstatic.cpp
applying patch /nix/store/s5myi2ym79i461fnzz9i25sj2ync4pc8-dlopen-gtkstyle.diff
patching file src/gui/styles/qgtkstyle_p.cpp
Hunk #3 succeeded at 605 (offset -3 lines).
applying patch /nix/store/inv66igmbwsr9vq0kg3npbwz1zqvy2l8-fix-medium-font.patch
patching file src/gui/text/qfontdatabase_x11.cpp
applying patch /nix/store/c2nqm32ihiyd08fw7dkqw661yn58vsrx-qt4-gcc6.patch
patching file configure
Hunk #1 succeeded at 7734 (offset -10 lines).
Hunk #2 succeeded at 8050 (offset -10 lines).
patching file src/xmlpatterns/api/qcoloroutput_p.h
configuring
fixing libtool script ./src/3rdparty/freetype/builds/unix/ltmain.sh
configure flags: -prefix /nix/store/m5fj1pd1jw7l35ambd4kcld8ci870rii-qt-mod-4.8.7 -dbus-linked -glib -no-separate-debug-info -openssl-linked -qdbus -v -confirm-license -exceptions -fast -graphicssystem raster -iconv -largefile -no-3dnow -no-accessibility -no-audio-backend -no-avx -no-cups -no-dbus -no-declarative -no-glib -no-gstreamer -no-gtkstyle -no-icu -no-javascript-jit -no-libmng -no-libtiff -nomake demos -nomake docs -nomake examples -nomake tests -nomake tools -nomake translations -no-mitshm -no-mmx -no-multimedia -no-nas-sound -no-neon -no-nis -no-opengl -no-openvg -no-pch -no-phonon -no-phonon-backend -no-qt3support -no-rpath -no-scripttools -no-sm -no-sql-ibase -no-sql-mysql -no-sql-odbc -no-sql-psql -no-sql-sqlite -no-sql-sqlite2 -no-sse -no-sse2 -no-sse3 -no-sse4.1 -no-sse4.2 -no-ssse3 -no-stl -no-xcursor -no-xfixes -no-xinerama -no-xinput -no-xkb -no-xrandr -no-xshape -no-xsync -opensource -release -static -system-libjpeg -system-libpng -system-zlib -webkit -xmlpatterns -docdir /nix/store/m5fj1pd1jw7l35ambd4kcld8ci870rii-qt-mod-4.8.7/share/doc/qt-4.8.7 -plugindir /nix/store/m5fj1pd1jw7l35ambd4kcld8ci870rii-qt-mod-4.8.7/lib/qt4/plugins -importdir /nix/store/m5fj1pd1jw7l35ambd4kcld8ci870rii-qt-mod-4.8.7/lib/qt4/imports -examplesdir /private/tmp/nix-build-qt-mod-4.8.7.drv-0/share/doc/qt-4.8.7/examples -demosdir /private/tmp/nix-build-qt-mod-4.8.7.drv-0/share/doc/qt-4.8.7/demos -datadir /nix/store/m5fj1pd1jw7l35ambd4kcld8ci870rii-qt-mod-4.8.7/share/qt-4.8.7 -translationdir /nix/store/m5fj1pd1jw7l35ambd4kcld8ci870rii-qt-mod-4.8.7/share/qt-4.8.7/translations
Determining system architecture... (Darwin:14.5.0:x86_64)
    'macosx' is supported
System architecture: 'macosx'
DWARF2 debug symbols disabled.
Xarch support detected
clang-4.0: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]
Symbol visibility control enabled.

This is the Qt for Mac OS X Open Source Edition.

...

/tmp/nix-build-qt-mod-4.8.7.drv-0/source/tools/shared -DQT_NO_PCRE -DQT_BUILD_QMAKE -DQT_BOOTSTRAPPED -DQLIBRARYINFO_EPOCROOT -DQT_NO_TEXTCODEC -DQT_NO_UNICODETABLES -DQT_NO_COMPONENT -DQT_NO_STL -DQT_NO_COMPRESS -I/private/tmp/nix-build-qt-mod-4.8.7.drv-0/source/mkspecs/macx-g++ -DHAVE_QCONFIG_CPP -DQT_NO_THREAD -DQT_NO_QOBJECT -DQT_NO_GEOM_VARIANT -DQT_NO_DEPRECATED  /private/tmp/nix-build-qt-mod-4.8.7.drv-0/source/src/corelib/kernel/qcore_mac.cpp
clang++ -o "/private/tmp/nix-build-qt-mod-4.8.7.drv-0/source/bin/qmake" project.o property.o main.o makefile.o unixmake2.o unixmake.o mingw_make.o option.o winmakefile.o projectgenerator.o meta.o makefiledeps.o metamakefile.o xmloutput.o pbuilder_pbx.o borland_bmake.o msvc_vcproj.o msvc_vcxproj.o msvc_nmake.o msvc_objectmodel.o msbuild_objectmodel.o symmake.o initprojectdeploy_symbian.o symmake_abld.o symmake_sbsv2.o symbiancommon.o registry.o epocroot.o gbuild.o qtextcodec.o qutfcodec.o qstring.o qtextstream.o qiodevice.o qmalloc.o qglobal.o qbytearray.o qbytearraymatcher.o qdatastream.o qbuffer.o qlist.o qfile.o qfilesystementry.o qfilesystemengine_unix.o qfilesystemengine.o qfilesystemiterator_unix.o qfsfileengine_unix.o qfsfileengine.o qfsfileengine_iterator.o qregexp.o qvector.o qbitarray.o qdir.o qdiriterator.o quuid.o qhash.o qfileinfo.o qdatetime.o qstringlist.o qabstractfileengine.o qtemporaryfile.o qmap.o qmetatype.o qsettings.o qsystemerror.o qlibraryinfo.o qvariant.o qvsnprintf.o qlocale.o qlocale_tools.o qlocale_unix.o qlinkedlist.o qnumeric.o qcryptographichash.o qxmlstream.o qxmlutils.o qsettings_mac.o qcore_mac.o -headerpad_max_install_names -framework ApplicationServices
ld: warning: object file (/nix/store/0gvrlmkb73hsyi1k5y1l1c6gjwslfja4-Libsystem-osx-10.11.6/lib/crt1.10.5.o) was built for newer OSX version (10.10) than being linked (10.5)
Undefined symbols for architecture x86_64:
  "__Unwind_Resume", referenced from:
      qmake_expandFunctions() in project.o
      void qmakeDeleteCacheClear<QMap<QString, ExpandFunc> >(void*) in project.o
      qmake_testFunctions() in project.o
      ParsableBlock::eval(QMakeProject*, QMap<QString, QStringList>&) in project.o
      QMakeProject::parse(QString const&, QMap<QString, QStringList>&, int) in project.o
      FunctionBlock::exec(QList<QStringList> const&, QMakeProject*, QMap<QString, QStringList>&, QStringList&) in project.o
      QList<QString>::clear() in project.o
      ...
ld: symbol(s) not found for architecture x86_64
clang-4.0: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:104: /private/tmp/nix-build-qt-mod-4.8.7.drv-0/source/bin/qmake] Error 1
builder for ‘/nix/store/j82a4pvyxxyfnx7pr34410f1w979cyxv-qt-mod-4.8.7.drv’ failed with exit code 2
cannot build derivation ‘/nix/store/rh8mksghdgksmq7w8xq8x8f0m190jf0a-wkhtmltopdf-0.12.4.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/rh8mksghdgksmq7w8xq8x8f0m190jf0a-wkhtmltopdf-0.12.4.drv’ failed
```

### Steps to reproduce

nix-env -i wkhtmltopdf

## Technical details

 - system: `"x86_64-darwin"`
 - host os: `Darwin 14.5.0, macOS 10.10.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.13`
 - channels(davidak): `"nixpkgs-18.03-darwin-18.03pre131588.ad027421c38"`
 - channels(root): `"nixpkgs-18.03pre121255.45a85eaceb"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

**Comment by knedlsepp** [comment]
I suspect some relation to #36947

**Comment by zimbatm** [comment]
fixed in #327641
